lib/spack/spack/cmd/diff.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import spack.cmd
+import spack.cmd.common.arguments as arguments
+import spack.environment as ev
+import spack.util.environment
+
+import llnl.util.tty as tty
+import spack.util.spack_json as sjson
+import operator
+import spack.cmd
+
+description = "compare two specs"
+section = "extensions"
+level = "short"
+
+
+def setup_parser(subparser):
+    """Parser is only constructed so that this prints a nice help
+       message with -h. """
+    arguments.add_common_arguments(
+        subparser, ['specs'])
+
+    subparser.add_argument(
+        '--json',
+        action='store_true',
+        default=False,
+        dest='dump_json',
+        help="Dump json output instead of pretty printing."
+    )
+    subparser.add_argument(
+        '--first',
+        action='store_true',
+        default=False,
+        dest='load_first',
+        help="load the first match if multiple packages match the spec"
+    )
+    subparser.add_argument(
+        '--diff-type',
```

Comment:
I think this should be called `--attributes` or something like that -- as for "diff type" I usually think side-by-side, multi-way, etc.